core-bundle/src/Image/Studio/ImageResult.php

```diff
@@ -154,25 +154,54 @@ public function getFilePath($absolute = false): string
      */
     public function createIfDeferred(): void
     {
+        $picture = $this->getPicture();
+
+        $array_flatten = static function (array $array): array {
+            return array_reduce($array, 'array_merge', []);
+        };
+
+        $findDeferredImages = static function (array $item) use ($array_flatten): array {
+            $candidates = array_merge(
+                [$item['src'] ?? null],
+                $array_flatten($item['srcset'] ?? [])
+            );
+
+            return array_filter(
+                $candidates,
+                static function ($image): bool {
+                    return $image instanceof DeferredImageInterface;
+                }
+            );
+        };
+
+        // Create an array of deferred images by looking through the 'src' and
+        // 'srcset' attributes of both the 'img' and all 'sources' tags.
+        $deferredImages = array_merge(
+            $findDeferredImages($picture->getImg()),
+            $array_flatten(array_map($findDeferredImages, $picture->getSources()))
+        );
+
+        if (empty($deferredImages)) {
+            return;
+        }
+
         $resizer = $this->resizer();
 
         if (!$resizer instanceof DeferredResizerInterface) {
             throw new \RuntimeException('The "contao.image.resizer" service does not support deferred resizing.');
         }
 
-        $picture = $this->getPicture();
+        $resizedPaths = [];
 
-        $deferredImages = array_filter(
-            array_column(
-                array_merge([$picture->getImg()], $picture->getSources()),
-                'src'
-            ),
-            static function ($image): bool {
-                return $image instanceof DeferredImageInterface;
+        /** @var DeferredImageInterface $deferredImage */
+        foreach ($deferredImages as $deferredImage) {
+            // Skip already processed images
+            if (isset($resizedPaths[$deferredImage->getPath()])) {
```

Comment:
I think we can skip this check. The deferred resizer is able to handle this case.

Reply:
see 4f30e2f9f8764de8f1ace572695d3c1e8cdb05a5